src/frontend/screens/Achievements/GameAchievementDetails/index.tsx

```diff
@@ -0,0 +1,158 @@
+import { GameAchievements } from '@hyperplay/ui'
+import { Achievement, AchievementSort, SummaryAchievement } from 'common/types'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { achievementsSortOptions } from '..'
+import { useMintAchievements } from '../MintAchievementsContext'
+import walletStore from 'frontend/store/WalletStore'
+import { useTranslation } from 'react-i18next'
+import { useAchievementStore } from '../AchievementStoreContext'
+
+const pageSize = 6
+
+function isTimestampInPast(unixTimestamp: number) {
+  const currentTime = new Date().getTime()
+  const timestampInMilliseconds = unixTimestamp * 1000 // Convert to milliseconds
+
+  return timestampInMilliseconds < currentTime
+}
+
+export default React.memo(function GameAchievementDetails(): JSX.Element {
+  const { t } = useTranslation()
+  const { id } = useParams()
+  const [summaryAchievement, setSummaryAchievement] =
+    useState<SummaryAchievement>()
+  const [achievementsData, setAchievementData] = useState<{
+    data: Achievement[]
+    currentPage: number
+    totalPages: number
+  }>({ data: [], currentPage: 0, totalPages: 0 })
+
+  const [selectedSort, setSelectedSort] = useState(achievementsSortOptions[0])
+
+  const {
+    achievementsToBeMinted,
+    isLoading,
+    handleMint,
+    handleUpdate,
+    achievementsToBeUpdated
+  } = useMintAchievements()
+  const { store, playerStoreId, numFreeMints } = useAchievementStore()
+
+  const fetchAchievements = useCallback(
+    async ({ page, sort }: { page: number; sort?: AchievementSort }) => {
+      return window.api.getIndividualAchievements({
+        gameId: Number(id),
+        store,
+        sort: sort ?? selectedSort.value,
+        page,
+        pageSize,
+        playerStoreId,
+        playerAddress: walletStore.address
+      })
+    },
+    [store, selectedSort, playerStoreId, walletStore.address, id]
+  )
```

Comment:
(see my comment below about mobx classes) 

api calls like this can be in a mobx class as well. these dependency arrays are not needed there because they are automatically determined